src/Common/Commands.Common/CustomAttributes/CmdletOutputBreakingChangeAttribute.cs

```diff
@@ -0,0 +1,98 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.WindowsAzure.Commands.Common.CustomAttributes
+{
+    [AttributeUsage(
+     AttributeTargets.Class,
+     AllowMultiple = true)]
+    public class CmdletOutputBreakingChangeAttribute : GenericBreakingChangeAttribute
+    {
+        public Type DeprecatedCmdLetOutputType { get; }
+
+        //This is still a String instead of a Type as this 
+        //might be undefined at the time of adding the attribute
+        public string ReplacementCmdletOutputTypeName { get; set; }
+
+        public string[] DeprecatedOutputProperties { get; set; }
+
+        public string[] NewOutputProperties { get; set; }
+
+        public CmdletOutputBreakingChangeAttribute(Type deprecatedCmdletOutputTypeName) :
+            base(string.Empty)
+        {
+            this.DeprecatedCmdLetOutputType = deprecatedCmdletOutputTypeName;
+        }
+
+        public CmdletOutputBreakingChangeAttribute(Type deprecatedCmdletOutputTypeName, string deprecateByVersion) :
+             base(string.Empty, deprecateByVersion)
+        {
+            this.DeprecatedCmdLetOutputType = deprecatedCmdletOutputTypeName;
+        }
+
+        public CmdletOutputBreakingChangeAttribute(Type deprecatedCmdletOutputTypeName, string deprecateByVersion, string changeInEfectByDate) :
+             base(string.Empty, deprecateByVersion, changeInEfectByDate)
+        {
+            this.DeprecatedCmdLetOutputType = deprecatedCmdletOutputTypeName;
+        }
+
+        protected override string GetAttributeSpecificMessage()
+        {
+            string message = null;
+
+            //check for the deprecation scenario
+            if (string.IsNullOrWhiteSpace(ReplacementCmdletOutputTypeName) && NewOutputProperties == null && DeprecatedOutputProperties == null && string.IsNullOrWhiteSpace(ChangeDescription))
+            {
+                message = string.Format(Resources.BreakingChangesAttributesCmdLetOutputTypeDeprecated, DeprecatedCmdLetOutputType.FullName);
+            }
+            else
+            {
+                if (!string.IsNullOrWhiteSpace(ReplacementCmdletOutputTypeName))
+                {
+                    message = string.Format(Resources.BreakingChangesAttributesCmdLetOutputChange1, DeprecatedCmdLetOutputType.FullName, ReplacementCmdletOutputTypeName);
+                }
+                else
+                {
+                    message = string.Format(Resources.BreakingChangesAttributesCmdLetOutputChange2, DeprecatedCmdLetOutputType.FullName);
+                }
+
+                if (DeprecatedOutputProperties != null && DeprecatedOutputProperties.Length > 0)
+                {
+                    message += Resources.BreakingChangesAttributesCmdLetOutputPropertiesRemoved + "\t";
+                    foreach (string propety in DeprecatedOutputProperties)
+                    {
+                        message += "'" + propety + "' ";
```

Comment:
use ```message=$"{message}, '{proeprty}'"```  I think you want separators here, and the string interpolation syntax is a bit easier to read